### PR TITLE
OpenAI GPT (Independent Publisher)

### PR DESCRIPTION
**Pull request by troystaylor**
image

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [X] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,652 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OpenAI",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json:4]: @troystaylor it is not allowed to have two connectors with the same name. is it possible to change the name? maybe to "OpenAI GPT-4"

File: independent-publisher-connectors/OpenAI GPT-4/apiProperties.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          "tooltip": "Provide your API Key",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/OpenAI GPT-4/apiProperties.json:18]: @troystaylor for all our independent publishers we ask to set the "iconBrandColor" to "#da3b01". Please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json).

File: independent-publisher-connectors/OpenAI GPT-4/apiProperties.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "scriptOperations": [],
+    "capabilities": [],
+    "policyTemplateInstances": [],
+    "publisher": "Troy Taylor"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/OpenAI GPT-4/apiProperties.json:22]: @troystaylor please add the stackOwner property.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
@troystaylor can you please provide some screenshots to confirm testing the connector?

File: independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,695 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OpenAI (V2)",
+    "description": "Connect to the OpenAI service and use the Power of GPT-4 and other models. API key must be entered as \"Bearer YOUR_API_KEY\"",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json:5]: @troystaylor can you please remove the word "API" from the description. you can add this information in the apiProperties in the description for the APIKey.
image

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json (hunk 4 above)
Comment (troystaylor) [reply independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json:5]: Yes, I'll remove it now, but that's the default description/tooltip created by the custom connector editor.

Comment (Amjed-Ayoub) [reply independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json:5]: @troystaylor thank you so much. I meant you can add this information in either the description or the tooltip for the APIKey :-)

**Comment by troystaylor** [comment]
@Amjed-Ayoub Can I get an update on where this is stuck being reviewed? I've been asked to work on the moderation endpoint, so I'd like to know if I can slide it in this PR later in the week or wait for a v1.1 after it has been deployed.

File: independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,695 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OpenAI (V2)",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json:4]: @troystaylor can you please change the title, our backend is not accepting this title as is. can we please make it "OpenAI GPT4" or something similar?

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json (hunk 5 above)
Comment (troystaylor) [reply independent-publisher-connectors/OpenAI GPT-4/apiDefinition.swagger.json:4]: Would OpenIA V2 work?

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @troystaylor,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @troystaylor ,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]
Hello @troystaylor ,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Pull request merged**